gravity-forms/gw-choice-counter.php

```diff
@@ -63,34 +63,42 @@ function output_script() {
 						}
 					}
 
-					self.init = function() {
-
-						// Event handler for all listeners to avoid DRY
-						var updateChoiceEventHandler = function() {
-							self.updateChoiceCount( self.formId, self.choiceFieldIds, self.countFieldId );
-						};
-
+					self.updateEventHandlers = function() {
 						for( var i = 0; i < self.choiceFieldIds.length; i++ ) {
 
 							var choiceFieldId       = self.choiceFieldIds[i],
 								choiceFieldSelector = '#input_' + self.formId + '_' + choiceFieldId,
 								$choiceField        = $(choiceFieldSelector),
 								$parentForm         = $choiceField.parents('form');
 
+							$parentForm.off( 'click', choiceFieldSelector, self.updateChoiceEventHander );
```

Comment:
Clean up existing event handlers

Reply:
I dig it!

One pro-tip here if getting a reference to the callback is impossible (or a lot of work) is to leverage jQuery event namespaces.

e.g. `$parentForm.on( 'click.gwChoiceCounter', choiceFieldSelector, self.updateChoiceEventHandler );`

then you can call...

e.g. `$parentForm.off( '.gwChoiceCounter' );`

Reply:
Smooooth :sunglasses: